pkg/backend/filestate/lock.go

```diff
@@ -79,6 +79,36 @@ func (b *localBackend) checkForLock(ctx context.Context, stackRef backend.StackR
 		}
 	}
 
+	return lockKeys, nil
+}
+
+// checkForLock looks for any existing locks for this stack, and returns a helpful diagnostic if there is one.
+func (b *localBackend) checkForLock(
+	ctx context.Context,
+	stackRef backend.StackReference,
+	lockTimeout time.Duration,
+) error {
+	lockKeys, err := b.getLockKeys(ctx, stackRef)
+	if err != nil {
+		return err
+	}
+
+	start := time.Now()
+
+	for len(lockKeys) > 0 && time.Since(start) < lockTimeout {
+		timeLeft := lockTimeout - time.Since(start)
+		fmt.Printf("\033[2K\rState is locked, waiting for lock... %s left", timeLeft.Truncate(time.Second))
```

Comment:
This should probably be sent via the diag.Sink object, not just a printf.

Reply:
can you post or link to some example code hwo to do that?

Reply:
Something like `b.d.Infoerrf(diag.Message("", "State is locked, waiting for lock... %s left"), timeLeft.Truncate(time.Second))`

Reply:
I think I figured it out

Reply:
Yeh that's unfortunate but we want to try and err away from printing directly to stdout/err in the engine. It's pretty deeply tied to display logic today, but the intention is to make a more delineated interface here.